assets/js/sidebar/sidebar-drawer.js

```diff
@@ -135,3 +157,21 @@ function closeSidebarIfSmallScreen () {
     closeSidebar()
   }
 }
+
+/**
+ * Track the sidebar preference for the user
+ */
+function setPreference () {
+  switch (state.sidebarPreference) {
+    case userPref.OPEN:
+      state.sidebarPreference = userPref.CLOSED
+      break
+    case userPref.CLOSED:
+      state.sidebarPreference = userPref.OPEN
```

Comment:
Is this correct? If the user preference is closed, we are setting it to open? 🤔

Reply:
The control logic is correct. The function is called when the `click` event tied to the toggle button fires. So whenever the user changes the state of the sidebar with the toggle button, we'll update their preference. 

> So, if you're preference _was_ closed we'll set it _open_
> 
> And if you're preference _was_ open, we'll set it to _closed_
> 
> And if you had _no_pref_, we'll set the pref to the state of the sidebar. 

I've attached a short video showing the state changes while working with the sidebar

https://user-images.githubusercontent.com/16793430/133001792-dbec7ca4-bf61-4c12-987b-7b5e0b4f7d60.MP4

Reply:
would it be better if I called it `togglePreference()` instead of `setPreference()`

Reply:
Beautiful video, thank you! 😍

Reply:
> would it be better if I called it togglePreference() instead of setPreference()

Ah, good idea! I just merged the PR though, but feel free to submit an update. :)

Reply:
will do